File: com/cc/java/binary_conversion.java
package com.cc.java;

public class binary_conversion {

    public static void main(String[] args) {
        System.out.print("Zahl (0-15) eingeben: ");
        String input = System.console() != null ? System.console().readLine() : new java.util.Scanner(System.in).nextLine();
        int number;
        try {
            number = Integer.parseInt(input);
        } catch (NumberFormatException e) {
            System.out.println("Ungültige Eingabe!");
            return;
        }
        if (number < 0 || number > 15) {
            System.out.println("Bitte eine Zahl zwischen 0 und 15 eingeben!");
            return;
        }
        StringBuilder binary = new StringBuilder();
        int n = number;
        do {
            binary.insert(0, n % 2);
            n = n / 2;
        } while (n > 0);
        System.out.println("Binär: " + binary.toString());
    }
}

File: com/cc/python/binary_conversion.py
user_input1 = input("Please enter an integer:")

def binary_conversion(user_input1):
    binary_digits = []
    
    if user_input1 > 15:
        return "Please enter a number from 0 - 15"
    
    for i in range(4):
        binary_digits.append(str(user_input1 % 2))
        user_input1 = user_input1 // 2

    binary_digits.reverse()
    return "The binary representation is: " + "".join(binary_digits)
print(binary_conversion(int(user_input1)))